### PR TITLE
With trigger

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: engine/api/trigger/trigger.go [hunk 1]
```diff
@@ -331,6 +331,60 @@ func LoadAutomaticTriggersAsSource(db database.Querier, appID, pipelineID, envID
 	return triggers, nil
 }
 
+// LoadTriggersByAppAndPipeline Load triggers for the given app and pipeline
+func LoadTriggersByAppAndPipeline(db database.Querier, appID int64, pipID int64) ([]sdk.PipelineTrigger, error) {
```
Comment (fsamin) [inline engine/api/trigger/trigger.go]: We should have tests for functions using funky queries and scanner functions

File: engine/api/application.go [hunk 2]
```diff
@@ -218,6 +219,18 @@ func getApplicationHandler(w http.ResponseWriter, r *http.Request, db *sql.DB, c
 		}
 	}
 
+	if withTriggers == "true" {
+		for i := range app.Pipelines {
+			appPip := &app.Pipelines[i];
```
Comment (fsamin) [inline engine/api/application.go]: why appPip is not defined by the range ?

Comment (sguiheux) [inline engine/api/application.go]: Need a pointer to set trigger on it just after

File: engine/api/application_test.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/ovh/cds/engine/api/application"
+	"github.com/ovh/cds/engine/api/auth"
+	"github.com/ovh/cds/engine/api/pipeline"
+	"github.com/ovh/cds/engine/api/sessionstore"
+	"github.com/ovh/cds/engine/api/testwithdb"
+	test "github.com/ovh/cds/engine/api/testwithdb"
```
Comment (fsamin) [inline engine/api/application_test.go]: double import

**Pull request merged**